ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7940,6 +7970,46 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
     return output;
   }
 
+  private Path getCTASDestinationTableLocation(CreateTableDesc tblDesc, boolean enableSuffixing) throws SemanticException {
+    Path location;
+    String suffix = "";
+    try {
+      // When location is specified, suffix is not added
+      if (tblDesc.getLocation() == null) {
```

Comment:
should we check the empty location as well?

Reply:
Location check happens now at the point where suffixing is decided. Updated.
https://github.com/apache/hive/pull/3281/files#diff-d4b1a32bbbd9e283893a6b52854c7aeb3e356a1ba1add2c4107e52901ca268f9R7616-R7618